### PR TITLE
Personalities innovations

### PR DESCRIPTION
**Pull request by FabianSimonGross**
Implemented the following issue: #182

Keep in mind we do not have all the data that is needed to produce the whole product in the mock ups.

### CODE REVIEW

File: rogue-thi-app/pages/index.js [hunk 1]
```diff
@@ -161,260 +147,33 @@ const ALL_DASHBOARD_CARDS = [
...
 
   return (
     <AppContainer>
       <AppNavbar title="neuland.app" showBack={false}>
-        <AppNavbar.Button onClick={() => setShowThemeModal(true)}>
-          <FontAwesomeIcon title="Personalisieren" icon={faPen} fixedWidth />
+        <AppNavbar.Button onClick={() => setShowPersonalizeModal(true)}>
+          <FontAwesomeIcon icon={faPen} title={'Personalisieren'} fixedWidth />
+        </AppNavbar.Button>
+        <AppNavbar.Button onClick={() => forgetSession(router)}>
```
Comment (alexhorn) [inline rogue-thi-app/pages/index.js:164]: Can you remove the logout button from the dashboard? I think the intention was to have it only on the profile page

File: rogue-thi-app/components/page/AppTabbar.js [hunk 2]
```diff
@@ -28,29 +29,35 @@ export default function AppTabbar () {
 
       <Navbar fixed="bottom" className={[styles.navbar, 'mobile-only']}>
         <Nav className={['justify-content-around', styles.nav]}>
+          <Nav.Item>
+            <Nav.Link onClick={() => router.replace('/library')} className={[styles.tab, router.pathname === '/library' && styles.tabActive]}>
+              <FontAwesomeIcon icon={faBook} className={styles.icon} />
+              Bibliothek
```
Comment (alexhorn) [inline rogue-thi-app/components/page/AppTabbar.js:35]: Can you move the tabbar changes to a future PR? I think that design element isn't finished yet

File: rogue-thi-app/components/modal/FilterFoodModal.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+            </h6>
+            <Button variant='outline-primary' onClick={() => {
+              setShowAllergenSelection(true)
+              setShowFoodFilterSelection(false)
+            }}>
+              <FontAwesomeIcon title="Allergene" icon={faPen} fixedWidth/>
+            </Button>
+          </div>
+          <div className={styles.filterBody}>
+            <>Ausgewählt:</>
```
Comment (alexhorn) [inline rogue-thi-app/components/modal/FilterFoodModal.js:80]: There should be a space after the colon:
image

File: rogue-thi-app/components/modal/FilterFoodModal.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+            <Button variant='outline-primary' onClick={() => {
+              setShowPreferencesSelection(true)
+              setShowFoodFilterSelection(false)
+            }}>
+              <FontAwesomeIcon title='Preferences' icon={faPen} fixedWidth/>
+            </Button>
+          </div>
+
+          <div className={styles.filterBody}>
+            <>Ausgewählt:</>
```
Comment (alexhorn) [inline rogue-thi-app/components/modal/FilterFoodModal.js:99]: There should be a space after the colon

File: rogue-thi-app/pages/personal.js [hunk 5]
```diff
@@ -34,61 +68,151 @@ export default function Personal () {
...
+          </ListGroup.Item>
+
+          <ListGroup.Item action onClick={() => window.open('https://outlook.thi.de/', '_blank')}>
+            <FontAwesomeIcon icon={faExternalLink} className={styles.interaction_icon}/>
+            E-Mail
+          </ListGroup.Item>
+
+          <ListGroup.Item action onClick={() => window.open('https://mythi.de', '_blank')}>
+            <FontAwesomeIcon icon={faExternalLink} className={styles.interaction_icon}/>
+            MyTHI (Beschäftigten Portal)
```
Comment (alexhorn) [inline rogue-thi-app/pages/personal.js:175]: Can we hide this from users that are not employees?

File: rogue-thi-app/pages/personal.js [hunk 6]
```diff
@@ -34,61 +68,151 @@ export default function Personal () {
...
+          <ListGroup.Item className="text-muted">
+            <span className={userdata ? styles.personal_value : styles.personal_value_loading}>
+            {userdata && userdata.stgru + '. Semester'}<br/>
             <a
               /* see: https://github.com/neuland-ingolstadt/THI-App/issues/90#issuecomment-924768749 */
               href={userdata?.po_url && userdata.po_url.replace('verwaltung-und-stabsstellen', 'hochschulorganisation')}
               target="_blank"
               rel="noreferrer">
-              {userdata.pvers}
+            Deine SPO <FontAwesomeIcon icon={faExternalLink}/>
```
Comment (alexhorn) [inline rogue-thi-app/pages/personal.js:103]: Do we need the SPO in this location? I feel like this is a feature that would be very rarely used

**Review by alexhorn — CHANGES_REQUESTED** [review]
There's a couple things that should be fixed, other than that it looks really good!

File: rogue-thi-app/package.json [hunk 7]
```diff
@@ -28,7 +28,8 @@
     "dompurify": "^2.3.4",
     "fetch-bypass-cors": "github:neuland-ingolstadt/fetch-bypass-cors#v1.0.3",
     "fetch-cookie": "^2.0.1",
-    "ical-generator": "^3.6.0",
+    "fs": "^0.0.1-security",
```
[the remark this answers is not in the document]
Comment (FabianSimonGross) [reply rogue-thi-app/package.json:31]: I haven't installed anything weird I'm gonna remove that and see if anything breaks

File: rogue-thi-app/styles/PersonalDataModal.module.css [hunk 8]
```diff
@@ -0,0 +1,20 @@
+.modalHeader h6 {
```
[the remark this answers is not in the document]
Comment (FabianSimonGross) [reply rogue-thi-app/styles/PersonalDataModal.module.css:1]: Done.

File: rogue-thi-app/styles/Personalize.module.css [hunk 9]
```diff
@@ -0,0 +1,24 @@
+.themeModal {
```
[the remark this answers is not in the document]
Comment (FabianSimonGross) [reply rogue-thi-app/styles/Personalize.module.css:1]: Done.

File: rogue-thi-app/styles/FilterFoodModal.module.css [hunk 10]
```diff
@@ -0,0 +1,11 @@
+.filterHeader h6 {
```
[the remark this answers is not in the document]
Comment (FabianSimonGross) [reply rogue-thi-app/styles/FilterFoodModal.module.css:1]: Done.

File: rogue-thi-app/components/modal/FilterFoodModal.js (hunk 3 above)
Comment (FabianSimonGross) [reply rogue-thi-app/components/modal/FilterFoodModal.js:80]: Yes.

File: rogue-thi-app/pages/personal.js (hunk 5 above)
Comment (FabianSimonGross) [reply rogue-thi-app/pages/personal.js:175]: Done.

File: rogue-thi-app/pages/index.js (hunk 1 above)
Comment (FabianSimonGross) [reply rogue-thi-app/pages/index.js:164]: Done. Removed the Theming Button also. As the Design suggested and we discussed.

File: rogue-thi-app/components/page/AppTabbar.js (hunk 2 above)
Comment (FabianSimonGross) [reply rogue-thi-app/components/page/AppTabbar.js:35]: Done.

File: rogue-thi-app/components/modal/FilterFoodModal.js (hunk 4 above)
Comment (FabianSimonGross) [reply rogue-thi-app/components/modal/FilterFoodModal.js:99]: Done.

File: rogue-thi-app/lib/hooks/food-filter.js [hunk 11]
```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState } from 'react'
+
+/**
+ * React Hook for Managing the Food Filter
+ * @returns {{setAllergenSelection: (value: (((prevState: {}) => {}) | {})) => void, allergenSelection: {}, toggleSelectedRestaurant: toggleSelectedRestaurant, preferencesSelection: {}, setPreferencesSelection: (value: (((prevState: {}) => {}) | {})) => void, selectedRestaurants: string[], isStudent: boolean, savePreferencesSelection: savePreferencesSelection, saveAllergenSelection: saveAllergenSelection}}
```
[the remark this answers is not in the document]
Comment (FabianSimonGross) [reply rogue-thi-app/lib/hooks/food-filter.js:5]: Done.

File: rogue-thi-app/lib/hooks/dashboard.js [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+  PLATFORM_DESKTOP,
+  PLATFORM_MOBILE,
+  USER_EMPLOYEE, USER_GUEST, USER_STUDENT
+} from '../../pages'
+import { useEffect, useState } from 'react'
+
+/**
+ * React Hook for Managing the Dashboard
+ * @returns {{
+ *   showDashboardEntries: object[]
```
[the remark this answers is not in the document]
Comment (FabianSimonGross) [reply rogue-thi-app/lib/hooks/dashboard.js:12]: Done.

**Review by alexhorn — APPROVED** [review]
Thank you!

**Pull request merged**